CrashLoopBackOff when defining ingress in spec.controlPlane

When specifying `spec.controlPlane.ingress` into a `TenantControlPlane` resource, kamaji controller gets into CrashLoopBackOff.

Environment:
kamaji version: v0.2.1
kamaji admin cluster version: v1.25.6
ingress controller: ingress-nginx

example tcp resource to test the issue with:
```
apiVersion: kamaji.clastix.io/v1alpha1
kind: TenantControlPlane
metadata:
  finalizers:
  - finalizer.kamaji.clastix.io
  - finalizer.kamaji.clastix.io/soot
  generation: 8
  name: kaas-tenant-001
  namespace: tenants
spec:
  addons:
    coreDNS: {}
    kubeProxy: {}
  controlPlane:
    deployment:
      additionalMetadata:
        labels:
          tenant.clastix.io: kaas-tenant-001
      extraArgs:
        apiServer:
        - --cloud-provider=external
        controllerManager:
        - --cloud-provider=external
      replicas: 1
      resources:
        apiServer:
          requests:
            cpu: 250m
            memory: 512Mi
        controllerManager:
          requests:
            cpu: 125m
            memory: 256Mi
        scheduler:
          requests:
            cpu: 125m
            memory: 256Mi
      strategy:
        rollingUpdate:
          maxSurge: 100%
          maxUnavailable: 0
        type: RollingUpdate
      ingress:
        additionalMetadata:
          annotations:
            nginx.ingress.kubernetes.io/ssl-passthrough: "true"
        hostname: kaas-tenant-001.example.com
    service:
      additionalMetadata:
        labels:
          tenant.clastix.io: kaas-tenant-001
      serviceType: ClusterIP
  dataStore: etcd
  kubernetes:
    admissionControllers:
    - ResourceQuota
    - LimitRanger
    kubelet:
      cgroupfs: systemd
      preferredAddressTypes:
      - InternalIP
      - ExternalIP
      - Hostname
    version: v1.26.1
  networkProfile:
    certSANs:
    - kaas-tenant-001.example.com
    dnsServiceIPs:
    - 10.96.0.10
    podCidr: 10.36.0.0/16
    port: 443
    serviceCidr: 10.96.0.0/16
```

Logs from kamaji controller:
```
2023/03/24 15:48:59 maxprocs: Updating GOMAXPROCS=1: using minimum allowed GOMAXPROCS
2023-03-24T15:48:59Z	INFO	setup	Kamaji version  
2023-03-24T15:48:59Z	INFO	setup	Build from: 
2023-03-24T15:48:59Z	INFO	setup	Build date: 
2023-03-24T15:48:59Z	INFO	setup	Go Version: go1.19.6
2023-03-24T15:48:59Z	INFO	setup	Go OS/Arch: linux/amd64
2023-03-24T15:49:00Z	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": ":8080"}
2023-03-24T15:49:00Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/migrate"}
2023-03-24T15:49:00Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "kamaji.clastix.io/v1alpha1, Kind=TenantControlPlane", "path": "/mutate-kamaji-clastix-io-v1alpha1-tenantcontrolplane"}
2023-03-24T15:49:00Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/mutate-kamaji-clastix-io-v1alpha1-tenantcontrolplane"}
2023-03-24T15:49:00Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "kamaji.clastix.io/v1alpha1, Kind=TenantControlPlane", "path": "/validate-kamaji-clastix-io-v1alpha1-tenantcontrolplane"}
2023-03-24T15:49:00Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/validate-kamaji-clastix-io-v1alpha1-tenantcontrolplane"}
2023-03-24T15:49:00Z	INFO	controller-runtime.builder	skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called	{"GVK": "/v1, Kind=Secret"}
2023-03-24T15:49:00Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "/v1, Kind=Secret", "path": "/validate--v1-secret"}
2023-03-24T15:49:00Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/validate--v1-secret"}
2023-03-24T15:49:00Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "kamaji.clastix.io/v1alpha1, Kind=DataStore", "path": "/mutate-kamaji-clastix-io-v1alpha1-datastore"}
2023-03-24T15:49:00Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/mutate-kamaji-clastix-io-v1alpha1-datastore"}
2023-03-24T15:49:00Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "kamaji.clastix.io/v1alpha1, Kind=DataStore", "path": "/validate-kamaji-clastix-io-v1alpha1-datastore"}
2023-03-24T15:49:00Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/validate-kamaji-clastix-io-v1alpha1-datastore"}
2023-03-24T15:49:00Z	INFO	setup	starting manager
2023-03-24T15:49:00Z	INFO	controller-runtime.webhook.webhooks	Starting webhook server
2023-03-24T15:49:00Z	INFO	controller-runtime.certwatcher	Updated current TLS certificate
2023-03-24T15:49:00Z	INFO	controller-runtime.webhook	Serving webhook server	{"host": "", "port": 9443}
2023-03-24T15:49:00Z	INFO	Starting server	{"path": "/metrics", "kind": "metrics", "addr": "[::]:8080"}
2023-03-24T15:49:00Z	INFO	Starting server	{"kind": "health probe", "addr": "[::]:8081"}
2023-03-24T15:49:00Z	INFO	controller-runtime.certwatcher	Starting certificate watcher
2023-03-24T15:49:16Z	DEBUG	events	kamaji-76b889b859-kgfhc_5bffd87e-69f7-4fef-8f90-3f6e92ca3227 became leader	{"type": "Normal", "object": {"kind":"Lease","namespace":"kamaji-system","name":"799b98bc.clastix.io","uid":"0e67f68a-748d-48dd-9aac-882772781277","apiVersion":"coordination.k8s.io/v1","resourceVersion":"381562"}, "reason": "LeaderElection"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "datastore", "controllerGroup": "kamaji.clastix.io", "controllerKind": "DataStore", "source": "kind source: *v1alpha1.DataStore"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "datastore", "controllerGroup": "kamaji.clastix.io", "controllerKind": "DataStore", "source": "kind source: *v1alpha1.TenantControlPlane"}
2023-03-24T15:49:16Z	INFO	Starting Controller	{"controller": "datastore", "controllerGroup": "kamaji.clastix.io", "controllerKind": "DataStore"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "kind source: *v1alpha1.TenantControlPlane"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "channel source: 0xc0000a0d70"}
2023-03-24T15:49:16Z	INFO	Starting Controller	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane"}
2023-03-24T15:49:16Z	INFO	Starting workers	{"controller": "datastore", "controllerGroup": "kamaji.clastix.io", "controllerKind": "DataStore", "worker count": 1}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "kind source: *v1alpha1.TenantControlPlane"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "kind source: *v1.Secret"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "kind source: *v1.ConfigMap"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "kind source: *v1.Deployment"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "kind source: *v1.Service"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "kind source: *v1.Ingress"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "channel source: 0xc0008c14a0"}
2023-03-24T15:49:16Z	INFO	Starting EventSource	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "source": "kind source: *v1.Job"}
2023-03-24T15:49:16Z	INFO	Starting Controller	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane"}
2023-03-24T15:49:16Z	INFO	Starting workers	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "worker count": 1}
2023-03-24T15:49:17Z	INFO	Starting workers	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "worker count": 1}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "validatingwebhookconfiguration", "controllerGroup": "admissionregistration.k8s.io", "controllerKind": "ValidatingWebhookConfiguration", "source": "kind source: *v1.ValidatingWebhookConfiguration"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "validatingwebhookconfiguration", "controllerGroup": "admissionregistration.k8s.io", "controllerKind": "ValidatingWebhookConfiguration", "source": "channel source: 0xc0000a09b0"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting Controller	{"controller": "validatingwebhookconfiguration", "controllerGroup": "admissionregistration.k8s.io", "controllerKind": "ValidatingWebhookConfiguration"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.ClusterRoleBinding"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.ServiceAccount"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.Role"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.RoleBinding"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.ConfigMap"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.DaemonSet"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "channel source: 0xc0000a1220"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting Controller	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "daemonset", "controllerGroup": "apps", "controllerKind": "DaemonSet", "source": "kind source: *v1.DaemonSet"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "daemonset", "controllerGroup": "apps", "controllerKind": "DaemonSet", "source": "kind source: *v1.ServiceAccount"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "daemonset", "controllerGroup": "apps", "controllerKind": "DaemonSet", "source": "kind source: *v1.ClusterRoleBinding"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "daemonset", "controllerGroup": "apps", "controllerKind": "DaemonSet", "source": "channel source: 0xc0000a0fa0"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting Controller	{"controller": "daemonset", "controllerGroup": "apps", "controllerKind": "DaemonSet"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "configmap", "controllerGroup": "", "controllerKind": "ConfigMap", "source": "kind source: *v1.ConfigMap"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "configmap", "controllerGroup": "", "controllerKind": "ConfigMap", "source": "channel source: 0xc0000a1f40"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting Controller	{"controller": "configmap", "controllerGroup": "", "controllerKind": "ConfigMap"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.ClusterRoleBinding"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.ClusterRole"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.ServiceAccount"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.Service"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.ConfigMap"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "kind source: *v1.Deployment"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding", "source": "channel source: 0xc0000a1860"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting Controller	{"controller": "clusterrolebinding", "controllerGroup": "rbac.authorization.k8s.io", "controllerKind": "ClusterRoleBinding"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "secret", "controllerGroup": "", "controllerKind": "Secret", "source": "kind source: *v1.Secret"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "secret", "controllerGroup": "", "controllerKind": "Secret", "source": "channel source: 0xc00040a0a0"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting Controller	{"controller": "secret", "controllerGroup": "", "controllerKind": "Secret"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "configmap", "controllerGroup": "", "controllerKind": "ConfigMap", "source": "kind source: *v1.ConfigMap"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting EventSource	{"controller": "configmap", "controllerGroup": "", "controllerKind": "ConfigMap", "source": "channel source: 0xc00040a000"}
2023-03-24T15:49:17Z	INFO	soot_tenants_kaas-tenant-001	Starting Controller	{"controller": "configmap", "controllerGroup": "", "controllerKind": "ConfigMap"}
2023-03-24T15:49:17Z	INFO	Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference	{"controller": "tenantcontrolplane", "controllerGroup": "kamaji.clastix.io", "controllerKind": "TenantControlPlane", "TenantControlPlane": {"name":"kaas-tenant-001","namespace":"tenants"}, "namespace": "tenants", "name": "kaas-tenant-001", "reconcileID": "ac906419-dec6-4918-aada-f5a38fc9fc90"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1ab7a62]

goroutine 491 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x1d85b80, 0x3445120})
	/usr/local/go/src/runtime/panic.go:884 +0x212
github.com/clastix/kamaji/internal/resources.(*KubernetesIngressResource).ShouldStatusBeUpdated(0xc00065d350?, {0x23c18b8?, 0xc0005844e0?}, 0x7?)
	/workspace/internal/resources/k8s_ingress_resource.go:29 +0x22
github.com/clastix/kamaji/internal/resources.createOrUpdate({0x23c18b8, 0xc0005844e0}, {0x23c77c0, 0xc00065d350}, 0xc000601800?)
	/workspace/internal/resources/resource.go:97 +0x94
github.com/clastix/kamaji/internal/resources.Handle({0x23c18b8, 0xc0005844e0}, {0x23c77c0?, 0xc00065d350?}, 0x0?)
	/workspace/internal/resources/resource.go:67 +0x10d
github.com/clastix/kamaji/controllers.(*TenantControlPlaneReconciler).Reconcile(0xc00078c000, {0x23c18b8, 0xc0005844e0}, {{{0xc000468496?, 0x7?}, {0xc0004684a0?, 0xf?}}})
	/workspace/controllers/tenantcontrolplane_controller.go:169 +0xd85
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x23c18b8?, {0x23c18b8?, 0xc0005844e0?}, {{{0xc000468496?, 0x1c8b800?}, {0xc0004684a0?, 0x0?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0005ca8c0, {0x23c1810, 0xc00025b100}, {0x1e2d6e0?, 0xc0003e3820?})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:323 +0x38f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0005ca8c0, {0x23c1810, 0xc00025b100})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.0/pkg/internal/controller/controller.go:231 +0x333
```
Reading the logs it seems that the issue happens during the tcp status update in the section `status.kubernetesResources.ingress`.

--- a/config/samples/kamaji_v1alpha1_tenantcontrolplane.yaml
+++ b/config/samples/kamaji_v1alpha1_tenantcontrolplane.yaml
@@ -1,22 +1,22 @@
 apiVersion: kamaji.clastix.io/v1alpha1
 kind: TenantControlPlane
 metadata:
-  name: test
+  name: k8s-126
 spec:
   controlPlane:
     deployment:
-      replicas: 1
+      replicas: 2
     service:
       serviceType: LoadBalancer
   kubernetes:
-    version: "v1.25.4"
+    version: "v1.26.0"
     kubelet:
-      cgroupfs: cgroupfs
-    admissionControllers:
-      - ResourceQuota
-      - LimitRanger
+      cgroupfs: systemd
   networkProfile:
     port: 6443
   addons:
     coreDNS: {}
     kubeProxy: {}
+    konnectivity:
+      server:
+        port: 8132

--- a/internal/resources/k8s_ingress_resource.go
+++ b/internal/resources/k8s_ingress_resource.go
@@ -25,13 +25,25 @@ type KubernetesIngressResource struct {
 	Name     string
 }
 
-func (r *KubernetesIngressResource) ShouldStatusBeUpdated(_ context.Context, tenantControlPlane *kamajiv1alpha1.TenantControlPlane) bool {
-	return !(tenantControlPlane.Status.Kubernetes.Ingress.Name == r.resource.GetName() &&
-		tenantControlPlane.Status.Kubernetes.Ingress.Namespace == r.resource.GetNamespace())
+func (r *KubernetesIngressResource) ShouldStatusBeUpdated(_ context.Context, tcp *kamajiv1alpha1.TenantControlPlane) bool {
+	// No update in case of no ingress in spec, neither in status.
+	if tcp.Spec.ControlPlane.Ingress == nil && tcp.Status.Kubernetes.Ingress == nil {
+		return false
+	}
+	// Must be updated when TCP is using an Ingress, and status is not tracking it.
+	if tcp.Spec.ControlPlane.Ingress != nil && tcp.Status.Kubernetes.Ingress == nil {
+		return true
+	}
+	// Must be updated when the status is referring to an Ingress, although spec doesn't.
+	if tcp.Spec.ControlPlane.Ingress == nil && tcp.Status.Kubernetes.Ingress != nil {
+		return true
+	}
+	// In case of status skew, update it.
+	return tcp.Status.Kubernetes.Ingress.Name != r.resource.GetName() || tcp.Status.Kubernetes.Ingress.Namespace != r.resource.GetNamespace()
 }
 
-func (r *KubernetesIngressResource) ShouldCleanup(tenantControlPlane *kamajiv1alpha1.TenantControlPlane) bool {
-	return tenantControlPlane.Spec.ControlPlane.Ingress == nil
+func (r *KubernetesIngressResource) ShouldCleanup(tcp *kamajiv1alpha1.TenantControlPlane) bool {
+	return tcp.Spec.ControlPlane.Ingress == nil
 }
 
 func (r *KubernetesIngressResource) CleanUp(ctx context.Context, _ *kamajiv1alpha1.TenantControlPlane) (bool, error) {
@@ -52,14 +64,16 @@ func (r *KubernetesIngressResource) CleanUp(ctx context.Context, _ *kamajiv1alph
 
 func (r *KubernetesIngressResource) UpdateTenantControlPlaneStatus(_ context.Context, tenantControlPlane *kamajiv1alpha1.TenantControlPlane) error {
 	if tenantControlPlane.Spec.ControlPlane.Ingress != nil {
-		tenantControlPlane.Status.Kubernetes.Ingress.IngressStatus = r.resource.Status
-		tenantControlPlane.Status.Kubernetes.Ingress.Name = r.resource.GetName()
-		tenantControlPlane.Status.Kubernetes.Ingress.Namespace = r.resource.GetNamespace()
+		tenantControlPlane.Status.Kubernetes.Ingress = &kamajiv1alpha1.KubernetesIngressStatus{
+			IngressStatus: r.resource.Status,
+			Name:          r.resource.GetName(),
+			Namespace:     r.resource.GetNamespace(),
+		}
 
 		return nil
 	}
 
-	tenantControlPlane.Status.Kubernetes.Ingress = &kamajiv1alpha1.KubernetesIngressStatus{}
+	tenantControlPlane.Status.Kubernetes.Ingress = nil
 
 	return nil
 }

--- a/internal/resources/k8s_ingress_resource.go
+++ b/internal/resources/k8s_ingress_resource.go
@@ -26,20 +26,25 @@ type KubernetesIngressResource struct {
 }
 
 func (r *KubernetesIngressResource) ShouldStatusBeUpdated(_ context.Context, tcp *kamajiv1alpha1.TenantControlPlane) bool {
-	// No update in case of no ingress in spec, neither in status.
-	if tcp.Spec.ControlPlane.Ingress == nil && tcp.Status.Kubernetes.Ingress == nil {
+	switch {
+	case tcp.Spec.ControlPlane.Ingress == nil && tcp.Status.Kubernetes.Ingress == nil:
+		// No update in case of no ingress in spec, neither in status.
 		return false
-	}
-	// Must be updated when TCP is using an Ingress, and status is not tracking it.
-	if tcp.Spec.ControlPlane.Ingress != nil && tcp.Status.Kubernetes.Ingress == nil {
+	case tcp.Spec.ControlPlane.Ingress != nil && tcp.Status.Kubernetes.Ingress == nil,
+		// Must be updated when TCP is using an Ingress, and status is not tracking it
+		// or
+		// Must be updated when the status is referring to an Ingress, although spec doesn't.
+		tcp.Spec.ControlPlane.Ingress == nil && tcp.Status.Kubernetes.Ingress != nil:
 		return true
-	}
-	// Must be updated when the status is referring to an Ingress, although spec doesn't.
-	if tcp.Spec.ControlPlane.Ingress == nil && tcp.Status.Kubernetes.Ingress != nil {
+	case len(r.resource.Status.LoadBalancer.Ingress) > 0 && tcp.Status.Kubernetes.Ingress == nil || tcp.Status.Kubernetes.Ingress.LoadBalancer.Ingress == nil:
+		// Must be updated since missing the Ingress status
 		return true
+	case r.resource.Status.LoadBalancer.Ingress[0].IP != tcp.Status.Kubernetes.Ingress.LoadBalancer.Ingress[0].IP:
+		// Must bne updated, Ingress load balancer IP is slightly different
+		return true
+	default:
+		return tcp.Status.Kubernetes.Ingress.Name != r.resource.GetName() || tcp.Status.Kubernetes.Ingress.Namespace != r.resource.GetNamespace()
 	}
-	// In case of status skew, update it.
-	return tcp.Status.Kubernetes.Ingress.Name != r.resource.GetName() || tcp.Status.Kubernetes.Ingress.Namespace != r.resource.GetNamespace()
 }
 
 func (r *KubernetesIngressResource) ShouldCleanup(tcp *kamajiv1alpha1.TenantControlPlane) bool {

